modules/basics/src/main/java/com/opengamma/strata/basics/market/MarketData.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.basics.market;
+
+import com.opengamma.strata.collect.timeseries.LocalDateDoubleTimeSeries;
+
+/**
+ * A set of market data used when calculating values for a single scenario.
+ */
+public interface MarketData {
+
+  /**
+   * Checks if this set of data contains a value for the specified key.
+   *
+   * @param key  a key identifying an item of market data
+   * @return true if this set of data contains a value for the specified key
+   */
+  public abstract boolean containsValue(MarketDataKey<?> key);
+
+  /**
+   * Checks if this set of data contains a time series for the specified key.
+   *
+   * @param key  a key identifying an item of market data
+   * @return true if this set of data contains a time series of market data for the specified key
+   */
+  public abstract boolean containsTimeSeries(ObservableKey key);
+
+  /**
+   * Returns a market data value.
+   * <p>
+   * The date of the market data is the same as the valuation date of the calculations.
+   *
+   * @param <T>  type of the market data
+   * @param key  key identifying the market data
+   * @return a market data value
+   * @throws IllegalArgumentException if there is no value for the specified key
+   */
+  public abstract <T> T getValue(MarketDataKey<T> key);
+
+  /**
+   * Returns a time series of market data values.
+   *
+   * @param key  key identifying the market data
+   * @return a time series of market data values
+   * @throws IllegalArgumentException if there is no time series for the specified key
+   */
+  public abstract LocalDateDoubleTimeSeries getTimeSeries(ObservableKey key);
+
+  /**
+   * Returns a mutable builder for building instances of {@code MarketData}.
+   *
+   * @return a mutable builder for building instances of {@code MarketData}
+   */
+  public static MarketDataBuilder builder() {
+    return new MarketDataBuilder();
+  }
```

Comment:
It was a conscious decision to put it last. The core functionality of the interface is provided by the non-static methods so those are first.